### ISSUE TITLE
VAAPI Transcoding ffmpeg version (Docker)

### ISSUE THREAD

**Issue by Pilo11** [jellyfin/jellyfin#266]
Hey guys.

I pulled your amazing docker image (jellyfin/jellyfin) and run it on my device. It works just fine. I configured VAAPI on Emby for better transcoding performance and CPU usage reducing. In the past (with the original Emby) this worked perfectly.

Now I'm discovering one problem when using VAAPI and transcoding. 
The log shows:

`/usr/bin/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"/mnt/videos/XXX.mpg" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 h264_vaapi  -b:v 1276000 -maxrate 1276000 -bufsize 2552000 -force_key_frames "expr:gte(t,n_forced*3)" -vf "format=nv12|vaapi,hwupload,deinterlace_vaapi" -copyts -vsync -1 -codec:a:0 aac -strict experimental -ac 2 -ab 224000  -f segment -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -segment_time 3  -individual_header_trailer 0 -segment_format mpegts -segment_list_type m3u8 -segment_start_number 0 -segment_list "/config/transcoding-temp/337f56616b3d8e1ec8d38f27ecbe34c1.m3u8" -y "/config/transcoding-temp/337f56616b3d8e1ec8d38f27ecbe34c1%d.ts"`

`...`

`2018-12-24 00:23:45.757 Info App: FFMpeg exited with code 1
2018-12-24 00:23:45.834 Info App: returning /config/transcoding-temp/337f56616b3d8e1ec8d38f27ecbe34c10.ts
...`

So I decided to run the ffmpeg command manually within a docker console and got this message:
`...
[AVFilterGraph @ 0x55727fdfdf20] No such filter: 'deinterlace_vaapi'
Error opening filters!`

I'm not an FFMPEG expert, so I decided to remove the filter from the long command text for trying it without this filter. And it worked...
I think this happens because the docker container has an old FFMPEG version, am I right or is it just something different?
The docker console says:
`ffmpeg version 3.2.12-1~deb9u1 Copyright (c) 2000-2018 the FFmpeg developers`

What are my possibilities in this case?
I've already forked this repo and removed the filter from the command line call: https://github.com/Pilo11/jellyfin/blob/66e2053dc6e79e1051a515152f1c0e3dc718612e/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L1721

I know this is not a real solution. I also tried to rebuild a docker container for me with the master Dockerfile but it failed with the "dotnet publish" command with a exit 1 code without any real error message (only build warnings...). But this is another story.

Maybe someone has a good idea what to do. Should we use another FFMPEG version in the docker container or is it an option to remove this filter?

**Comment by JustAMan** [comment]
My idea of solving this particular dependency hell is #191, you can follow that. VAAPI is the first on the list for me as it's the one I personally am using now :)

**Comment by Pilo11** [comment]
This issue is fixed with the latest changes from nvllsvm in the Dockerfile for using static ffmpeg deb in the docker container.

**Comment by Pilo11** [comment]
https://github.com/jellyfin/jellyfin/pull/728

**Comment by Pilo11** [comment]
#728

**Comment by fightforlife** [comment]
So a quick question, since I also want to use vaapi.
I pulled the current Master-Branch Docker image of jellyfin and activated vaapi. But hw acceleration is not even called in the ffmpg commands.

So I guess I have to fork jellyfin and change the following line;
FROM jrottenberg/ffmpeg:4.0-scratch as ffmpeg
to
FROM jrottenberg/ffmpeg:4.0-vaapi as ffmpeg
for Vaapi to work?

Thanks in advance!

**Comment by JustAMan** [comment]
cc @nvllsvm

**Comment by Pilo11** [comment]
> So a quick question, since I also want to use vaapi.
> I pulled the current Master-Branch Docker image of jellyfin and activated vaapi. But hw acceleration is not even called in the ffmpg commands.
> 
> So I guess I have to fork jellyfin and change the following line;
> FROM jrottenberg/ffmpeg:4.0-scratch as ffmpeg
> to
> FROM jrottenberg/ffmpeg:4.0-vaapi as ffmpeg
> for Vaapi to work?
> 
> Thanks in advance!

Correct. That's the way it works for me :D

**Comment by fightforlife** [comment]
Thank you! 
In the meantime I had an Idea that could be even easier using Docker compose.

One could run both images in parallel and mount the ffmpg executable from the ffmpg-vaapi container inside the jellyfin container. This way you dont need to fork anything.

Sadly this isnt working for me right know (see example below)
Jellyfin sys there is not ffmpg executable inside the "ffmpg_host" folder even if I can see the file in the jellyfin webinterface!

Any Ideas?

```
version: "2"

services:
  jellyfin:
    image: jellyfin/jellyfin:master
    ports:
      - "9025:8096/tcp"
    restart: always
    volumes:
      - jellyfin_config:/config
      - ffmpg_bin:/ffmpg_host
      - /mnt/HDD1:/mnt/HDD1
    devices:
      - /dev/dri/renderD128:/dev/dri/renderD128
    environment:
      - UID=1000
      - GID=1000
      - GIDLIST=44
  ffmpg:
    image: jrottenberg/ffmpeg:4.0-vaapi
    restart: always
    volumes:
      - ffmpg_bin:/usr
```

**Comment by JLFN** [comment]
> Thank you!
> In the meantime I had an Idea that could be even easier using Docker compose.
> 
> One could run both images in parallel and mount the ffmpg executable from the ffmpg-vaapi container inside the jellyfin container. This way you dont need to fork anything.
> 
> Sadly this isnt working for me right know (see example below)
> Jellyfin sys there is not ffmpg executable inside the "ffmpg_host" folder even if I can see the file in the jellyfin webinterface!
> 
> Any Ideas?
> 
> ```
> version: "2"
> 
> services:
>   jellyfin:
>     image: jellyfin/jellyfin:master
>     ports:
>       - "9025:8096/tcp"
>     restart: always
>     volumes:
>       - jellyfin_config:/config
>       - ffmpg_bin:/ffmpg_host
>       - /mnt/HDD1:/mnt/HDD1
>     devices:
>       - /dev/dri/renderD128:/dev/dri/renderD128
>     environment:
>       - UID=1000
>       - GID=1000
>       - GIDLIST=44
>   ffmpg:
>     image: jrottenberg/ffmpeg:4.0-vaapi
>     restart: always
>     volumes:
>       - ffmpg_bin:/usr
> ```

This worked for me 
```.yaml
version: "2.4"
services:
  jellyfin:
    container_name: jellyfin
    image: jellyfin/jellyfin
    restart: always
    network_mode: host
    mem_limit: 4096m
    cpu_percent: 50
    cpus: 3.0
    privileged: true
    environment:
      TZ: Europe/Stockholm
      # PGID: 1000
      # PUID: 1000
    volumes:
      - /home/USER$/.jellyfin/config:/config
      - /home/USER$/.jellyfin/cache:/cache
      - /home/USER$/.jellyfin/media:/media
      - /media/dkns/:/media/dkns
      - /home/homeassistant/dehydrated/certs/mydomain.duckdns.org:/certs/
    devices:
      - /dev/dri:/dev/dri
    ports:
      - "8096:8096"
      - "8095:8095"
      - "8921:8921"
      - "8920:8920"
      - "1900:1900"
#openssl pkcs12 -export -out /home/homeassistant/dehydrated/certs/mydomain.duckdns.org/mydomain.duckdns.org.pfx -inkey privkey.pem -in cert.pem -certfile chain.pem -password pass:myhardtocrackpassword
#http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration
#https://docs.docker.com/compose/compose-file/compose-file-v2/#cpu-and-other-resources
#https://github.com/docker/compose/releases/
#docker logs -f jellyfin
#docker exec -it jellyfin usr/bin/env bash
#make it work with emby lg webostv app #https://github.com/jellyfin/jellyfin/issues/588
#view-source:https://mydomain.duckdns.org:8920/web/index.html#!/home.html
```
